src/main/java/com/thealgorithms/bitmanipulation/HighestSetBit.java

```diff
@@ -0,0 +1,30 @@
+package com.thealgorithms.bitmanipulation;
+import java.util.Optional;
+
+/**
+ * Find Highest Set Bit
+ * @author Bama Charan Chhandogi (https://github.com/BamaCharanChhandogi)
+ */
+
+public final class HighestSetBit {
+    private HighestSetBit() {
+    }
+
+    public final static Optional<Integer> findHighestSetBit(int num) {
+        if (num < 0) {
+            throw new IllegalArgumentException("Input cannot be negative");
+        }
+
+        if (num == 0) {
+            return Optional.empty();
+        }
+
+        int position = 0;
+        while (num > 0) {
+            num >>= 1;
+            position++;
+        }
+
+        return Optional.of(position - 1);
+    }
+}
```

Comment:
```suggestion
}

```
After merging #4343, not having a _new line symbol_ at the end of the file will cause [Clang format linter](https://github.com/TheAlgorithms/Java/blob/09950d6097ea7b2590fb80336d144ee402e61226/.github/workflows/clang-format-lint.yml) to fail.

In any case, please _rebase_ before merging.

Reply:
Done